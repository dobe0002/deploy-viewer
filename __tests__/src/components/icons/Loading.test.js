/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Loading from '../../../../src/components/icons/Loading';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('Loading icon component tests', () => {
  test('Accessibility check', async () => {
    const wrapper = mount(
      <main>
        <Loading />
      </main>
    ); // Main tag is needed to prevent landmark errors
    const results = await axe(wrapper.getDOMNode());
    // console.log('Axe violations', results.violations);
    expect(results).toHaveNoViolations();
    // console.log(wrapper.html())
  });
  test('Loading is SVG based', () => {
    const loading = shallow(<Loading />);
    expect(loading.find('svg')).toHaveLength(1);
  });
  test('Snapshot test', () => {
    const loading = shallow(<Loading />);
    expect(loading).toMatchSnapshot();
  });
});
