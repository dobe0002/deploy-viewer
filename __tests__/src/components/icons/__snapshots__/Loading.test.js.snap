// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading icon component tests Snapshot test 1`] = `
<svg
  aria-label="Loading"
  id="svg-loader-1"
  role="img"
  style={
    Object {
      "height": "60px",
      "width": "60px",
    }
  }
  version="1.1"
  viewBox="0 0 80 80"
  x="0px"
  xmlSpace="preserve"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <path
    d="M40,72C22.4,72,8,57.6,8,40C8,22.4,22.4,8,40,8c17.6,0,32,14.4,32,32c0,1.1-0.9,2-2,2s-2-0.9-2-2c0-15.4-12.6-28-28-28S12,24.6,12,40s12.6,28,28,28c1.1,0,2,0.9,2,2S41.1,72,40,72z"
    fill="#575eff"
    id="spinner"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      dur="1.0s"
      from="0 40 40"
      repeatCount="indefinite"
      to="360 40 40"
      type="rotate"
    />
  </path>
</svg>
`;
